src/hooks/useSubgraphTokenRecords.ts

```diff
@@ -28,65 +35,107 @@ import {
+    filter: {
+      ...baseFilter,
+    },
+    recordCount: DEFAULT_RECORD_COUNT,
+    endpoint: endpointNotNull.current,
+  });
+  const queryOptions = useRef<QueryOptionsType>({
+    enabled: false,
+    getNextPageParam: paginator.current,
+  });
+  // Handle changes to props and re-generate cached variables
+  // These setter calls are co-located to avoid race conditions that can result in strange behaviour (OlympusDAO/olympus-frontend#2325)
+  useEffect(() => {
+    console.info(`${functionName}: Inputs changed. Updating calculated values`);
+
+    // We need to wipe the data, otherwise it will be inconsistent
+    // This is called here so that calling components can be updated before any changes to query configuration
+    setByDateTokenRecords(null);
+
+    const _finishDate = getISO8601String(adjustDateByDays(new Date(), 1)); // Tomorrow
+    queryVariables.current = {
       filter: {
         ...baseFilter,
-        date_gte: initialStartDate,
-        date_lt: initialFinishDate,
+        date_gte: !earliestDate ? null : getNextPageStartDate(_finishDate, earliestDate, dateOffset),
+        date_lt: _finishDate,
       },
       recordCount: DEFAULT_RECORD_COUNT,
-      endpoint: subgraphUrl,
-    },
-    {
-      enabled: earliestDate !== null && baseFilter != null,
+      endpoint: endpointNotNull.current,
+    };
+
+    dataSource.current = {
+      endpoint: endpointNotNull.current,
+    };
+
+    // Create a new paginator with the new earliestDate
+    const tempPaginator =
+      earliestDate !== null && subgraphUrl !== null
+        ? getNextPageParamFactory(chartName, earliestDate, DEFAULT_RECORD_COUNT, baseFilter, subgraphUrl, dateOffset)
+        : undefined;
+    paginator.current = tempPaginator;
+
+    queryOptions.current = {
+      enabled: earliestDate !== null && endpointNotNull.current.length > 0 && paginator.current !== undefined,
       getNextPageParam: paginator.current,
-    },
+    };
+  }, [baseFilter, earliestDate, dateOffset, endpointNotNull, functionName, subgraphUrl, chartName]);
+
+  /**
+   * Data fetching
+   */
+  // Fetch data (with included pagination)
+  const { data, hasNextPage, fetchNextPage, refetch, isFetching } = useInfiniteTokenRecordsQuery(
+    dataSource.current,
+    "filter",
+    queryVariables.current,
+    queryOptions.current,
   );
 
-  // Handle date changes
+  /**
+   * If the queryOptions change (triggered by the props changing), then we will force a refetch.
+   */
   useEffect(() => {
-    // We can't create the paginator until we have an earliestDate
-    if (!earliestDate || !baseFilter) {
-      return;
-    }
-
-    console.info(`${functionName}: earliestDate changed to ${earliestDate}. Re-fetching.`);
-
-    // We need to wipe the data, otherwise it will be inconsistent
-    setByDateTokenRecords(null);
+    // refetch does not respect the enabled property in react-query, so we check the enabled property
+    if (!queryOptions.current.enabled) return;
```

Comment:
Hmm I never knew that. Nice approach

Reply:
It's in the documentation, but it's not very clear. I was wondering why the queries were firing, despite being disabled (e.g. null subgraph URL or null `earliestDate`), and traced it to this.